bot/exts/recruitment/talentpool/_review.py

```diff
@@ -78,14 +77,14 @@ def schedule_review(self, user_id: int) -> None:
 
     async def post_review(self, user_id: int, update_database: bool) -> None:
         """Format the review of a user and post it to the nomination voting channel."""
-        review, reviewed_emoji = await self.make_review(user_id)
+        review, reviewed_emoji, nominee = await self.make_review(user_id)
         if not review:
             return
 
         guild = self.bot.get_guild(Guild.id)
         channel = guild.get_channel(Channels.nomination_voting)
 
-        log.trace(f"Posting the review of {user_id}")
+        log.trace(f"Posting the review of {nominee} ({nominee.id})")
```

Comment:
So! This is fun. The addition of `nominee.id` will no longer make this fail silently if a user leaves the guild after they were nominated but before they're up for a vote.

`self.make_review()` returns (string, None, None). `review` is therefore a non-empty string and that short-circuit logic there won't work anymore. `nominee.id` now tries to access the `id` attribute which a NoneType doesn't have.

So, the `if not review` should explicitly check for the nominee, not the review.

Reply:
Fixed in https://github.com/python-discord/bot/pull/1928/commits/769660b46b1eebedadeccc9d22a7bdf7705befc2